### PR TITLE
 belt immunity setting

### PR DESCRIPTION
**Pull request by 0ptera**
I'm using belts to automatically move planes into position for refueling.

Instead of just forking a version with belt_immunity = false I thought I'd properly add a setting. While doing so I also cleaned up duplicate entity prototypes and future proofed the code with a list of parameters per plane. ;)

Changelog:
- added belt immunity setting with en and de translation
- refactored plane creation script
- added mod setting order
- removed useless per_user flag from startup settings

### CODE REVIEW

File: locale/de/all.cfg [hunk 1]
```diff
@@ -67,11 +67,13 @@ aircraft-afterburner=Nachbrenner
...
 aircraft-sound-setting=Verwende Flugzeug Geräusche
 aircraft-hardmode=Gefahrenmodus
+aircraft-belt-immunity=Förderband Immunität
 helicopter-tech=Schlüsseltechnologie für Helicopter
 raven-tech=Schlüsseltechnologie für Raven
 
 [mod-setting-description]
 aircraft-sound-setting=Deaktivieren, um in allen Flugzeugen Fahrzeug- statt Flugzeug-Geräusche zu verwenden.
 aircraft-hardmode=Macht Flugzeuge anfälliger für Angriffe von Aliens. Benutzung auf eigene Gefahr!
+aircraft-belt-immunity=Föderbänder haben keinen einfluß auf Flugzeuge.
```
Comment (ST-DDT) [inline locale/de/all.cfg:77]: `einfluß` -> `Einfluss`

Comment (ST-DDT) [reply locale/de/all.cfg:77]: `Föderbänder` -> `Förderbänder`

File: settings.lua [hunk 2]
```diff
@@ -5,41 +5,48 @@ data:extend(
     name = "aircraft-sound-setting",
     setting_type = "startup",
     default_value = true,
-    per_user = false,
+    order="aa",
```
Comment (ST-DDT) [inline settings.lua:8]: formatting

File: settings.lua [hunk 3]
```diff
@@ -5,41 +5,48 @@ data:extend(
...
-	type = "bool-setting",
-	name = "aircraft-hardmode",
-	setting_type = "startup",
-	default_value = false,
-	per_user = false,
+    type = "bool-setting",
+    name = "aircraft-hardmode",
+    setting_type = "startup",
+    default_value = false,
+    order="ab",
```
Comment (ST-DDT) [inline settings.lua:15]: formatting

File: settings.lua [hunk 4]
```diff
@@ -5,41 +5,48 @@ data:extend(
...
-  type = "bool-setting",
-  name = "helicopter-tech",
-  setting_type = "startup",
-  default_value = false,
-  per_user = false,
+    type = "bool-setting",
+    name = "helicopter-tech",
+    setting_type = "startup",
+    default_value = false,
+    order="ba",
```
Comment (ST-DDT) [inline settings.lua:29]: formatting

File: settings.lua [hunk 5]
```diff
@@ -5,41 +5,48 @@ data:extend(
...
-  type = "bool-setting",
-  name = "raven-tech",
-  setting_type = "startup",
-  default_value = false,
-  per_user = false,
+    type = "bool-setting",
+    name = "raven-tech",
+    setting_type = "startup",
+    default_value = false,
+    order="bc",
```
Comment (ST-DDT) [inline settings.lua:36]: formatting

File: settings.lua [hunk 6]
```diff
@@ -5,41 +5,48 @@ data:extend(
...
   },
 --[[  {
   type = "bool-setting",
   name = "raven-equipment-grid",
   setting_type = "startup",
   default_value = false,
-  per_user = false,
+  order="bd",
```
Comment (ST-DDT) [inline settings.lua:43]: formatting

File: settings.lua [hunk 7]
```diff
@@ -5,41 +5,48 @@ data:extend(
...
-  type = "bool-setting",
-  name = "heli-equipment-grid",
-  setting_type = "startup",
-  default_value = false,
-  per_user = false,
+    type = "bool-setting",
+    name = "heli-equipment-grid",
+    setting_type = "startup",
+    default_value = false,  
+    order="bb",
```
Comment (ST-DDT) [inline settings.lua:50]: formatting and trailing whitespace

**Review by ST-DDT — CHANGES_REQUESTED** [review]
Looks like a good simplification.

-------------

Just some rambling from a random passerby:

On the other hand, I feel torn back and forth.
Formatting code is good, but during PRs you should generally update as few as possible to keep everything simple to review. Formatting can always be done in separate PRs.

And if you include formatting its very important that you stick to the formatting yourself as well.

**Comment by Stifling-Bossness** [comment]
We'll give this a shot and see how it turns out. The creation script looks simple enough to troubleshoot if it breaks.

**Pull request merged**